bin/fuel-core/src/cli/run.rs

```diff
@@ -266,25 +260,25 @@ impl Command {
             info!("Block production disabled");
         }
 
+        let consensus_key = load_consensus_key(consensus_key)?;
+        if consensus_key.is_some() && trigger == Trigger::Never {
+            warn!("Consensus key configured but block production is disabled!");
+        }
+
         // if consensus key is not configured, fallback to dev consensus key
-        let consensus_key = load_consensus_key(consensus_key)?.or_else(|| {
-            if consensus_dev_key && trigger != Trigger::Never {
+        let consensus_key = consensus_key.or_else(|| {
```

Comment:
I would prefer to merge that into one "if/else" statement, but it's okay to keep it like this